EIPS/eip-2774.md

```diff
@@ -0,0 +1,129 @@
+
+To resolve the issue, it should be possible to extend the deadline for time-based actions if there is a sudden spike in the network fee that may prevent price-sensitive users from participating. However, to extend the deadline, the blockchain must have a gas-price oracle that can self-inspect the blockchain and measure fees paid by its users. If a gas oracle can be implemented, then it is feasible to determine, on a per-block bassis, if the network fees have become *unreasonable* for some smart contract participants. As we outline in this EIP, we propose the ```BASEFEE``` from EIP-1559 can be used as a gas-price oracle and the fixed deadline should be replaced with a sliding window.
+
+To summarise our proposal, if at least ```K``` of the ```N``` previous blocks could have accepted transactions with a *reasonable* fee, then it is assumed the smart contract participants should have performed their action. 
+
+## Abstract
+
+We propose the first price-sensitive challenge process in the form of a new opcode ```relativeDeadline(): bool``` and it has two components.
+
+**Defining max fee.** We define the ```MAXFEE``` as the maximum gas price the user is willing to pay for their transaction to be included in the blockchain. We re-use the ```BASEFEE``` from EIP-1559 to determine the minimum gas price a transaction must include in order to be accepted into the respective block. As long as ```MAXFEE >= BASEFEE```, then we know the transaction could have been accepted into the block. 
+
+**Sliding time window.** We change the challenge time period from a fixed deadline ```T``` to a sliding window such that the blockchain checks if ```K``` of the previous ```N``` blocks can include transactions that paid at most ```MAXFEE```. If that condition is not satisfied, then the sliding window progresses until either 1) the ```BASEFEE``` comes down or 2) it reverts to a distant fixed deadline ```T```. 
+
+Note: If we assume the basefee is publicly accessible in a smart contract, then a new opcode is not required. However if it is implemented as a pre-compile, then the gas-costs for the lookup can potentially be cheaper. 
```

Comment:
Have you contacted anyone involved with EIP-1559 to determine if `BASEFEE` will be accessible in the EVM? If they are planning for it not to be, they may have a reason for doing so. Otherwise, adding that opcode would probably be better suited to occur in EIP-1559 itself.